src/main/java/org/folio/rest/impl/PermsAPI.java

```diff
@@ -274,167 +290,58 @@ public void putPermsUsersById(String id, PermissionUser entity, Map<String, Stri
-              PermissionUser permUser = permUsers.get(0);
-              try {
-                updateUserPermissions(connection, id,
-                    new JsonArray(permUser.getPermissions()), new JsonArray(),
-                    vertxContext, tenantId, okapiHeaders)
-                    .onComplete(updateUserPermsRes -> {
-                  if (updateUserPermsRes.failed()) {
-                    pgClient.rollbackTx(connection, rollback -> {
-                      String errStr = String.format("Error updating metadata: %s",
-                          updateUserPermsRes.cause().getMessage());
-                      logger.error(errStr, updateUserPermsRes.cause());
-                      asyncResultHandler.handle(Future.succeededFuture(
-                          DeletePermsUsersByIdResponse.respond500WithTextPlain(errStr)));
-                    });
-                    return;
-                  }
-                  pgClient.delete(connection, TABLE_NAME_PERMSUSERS,
-                      id, deleteReply -> {
-                        if (deleteReply.failed()) {
-                          pgClient.rollbackTx(connection, rollback -> {
-                            String errStr = String.format("Error deleting user: %s",
-                                deleteReply.cause().getMessage());
-                            logger.error(errStr, deleteReply.cause());
-                            asyncResultHandler.handle(Future.succeededFuture(
-                                DeletePermsUsersByIdResponse.respond500WithTextPlain(errStr)));
-                          });
-                          return;
-                        }
-                        if (deleteReply.result().rowCount() == 0) {
-                          pgClient.rollbackTx(connection, rollback ->
-                            asyncResultHandler.handle(Future.succeededFuture(
-                                DeletePermsUsersByIdResponse.respond404WithTextPlain(id)))
-                          );
-                          return;
-                        }
-                        pgClient.endTx(connection, done ->
-                          asyncResultHandler.handle(Future.succeededFuture(
-                              DeletePermsUsersByIdResponse.respond204()))
-                        );
-                      });
-                });
-              } catch (Exception e) {
-                //rollback
-                pgClient.rollbackTx(connection, rollback -> {
-                  String errStr = String.format("Error deleting user: %s",
-                      e.getMessage());
-                  logger.error(errStr, e);
-                  asyncResultHandler.handle(Future.succeededFuture(
-                      DeletePermsUsersByIdResponse.respond500WithTextPlain(errStr)));
-                });
-              }
-            })
-      );
+      pgClient
+          .withTrans(connection ->
+              connection.get(TABLE_NAME_PERMSUSERS, PermissionUser.class, idCrit, true)
```

Comment:
Using getByIdForUpdate avoids idCrit and locks the record to prevent a transaction rollback due to concurrent access.

Reply:
The getByIdForUpdate takes AsyncResult<SQLConnection> which smells like the old API. I'd have to start a transaction using the old API , because we have ther updateUserPermissions below that is part of this.

Reply:
`Conn connection` also has a getByIdForUpdate:
https://github.com/folio-org/raml-module-builder/blob/v33.1.1/domain-models-runtime/src/main/java/org/folio/rest/persist/Conn.java#L151-L160